fix(ivy): compile NgModule type providers without @Injectable annotation

pkozlowski-opensource opened the pull request:
Ref: FW-718

petebacondarwin commented on packages/upgrade/test/BUILD.bazel:
```diff
@@ -11,7 +11,6 @@ ts_library(
         "//packages/platform-browser",
         "//packages/platform-browser-dynamic",
         "//packages/platform-browser/testing",
-        "//packages/private/testing",
```
🥇

pkozlowski-opensource replied:
@petebacondarwin yep, this is the last test in `ngUpgrade` :-)

petebacondarwin reviewed (approved):
Woohoo! Last fixmeivy on upgrade??

pkozlowski-opensource commented:
@JoostK yeh, there are more cases where a type with dependencies and without `@Inject` annotation can be used and still "work" in the current view engine (at least in the JIT mode), ex. https://stackblitz.com/edit/angular-gsutee?file=src%2Fapp%2Fapp.module.ts 

I'm referring to:

```javascript
{
      provide: 'service',
      useClass: Ng2Service,     
}
```

Need to test if it works in the AOT mode today....

mhevery commented:
http://test/OCL:230809548:BASE:230810729:1548375474808:e52e7cde

picheli20 commented on packages/core/src/di/util.ts:
```diff
@@ -53,3 +53,9 @@ export function convertInjectableProviderToFactory(
+  const stringifiedTokens = dependencyTokens.map(token => token ? stringify(token) : '?');
+  return new Error(
+      `Can't resolve all parameters for ${stringify(type)}: (${stringifiedTokens.join(', ')}).`);
+}
```
Just a detail, the new line on the file end :)

https://github.com/angular/angular/blob/master/.editorconfig#L10

alxhub commented:
(@pkozlowski-opensource and I talked about this over Slack, this is a summary of our conclusions)

While this solution would work for JIT, there's no way for the AOT compiler to support it. DI in Ivy is a runtime concept, and the AOT compiler does not introspect the providers passed to any `@NgModule`. In fact, it's impossible to do so - if you have `providers` which reference a `ModuleWithProviders` function from another module, or a constant array from another module, we don't have the information at build time.

So the correct solution here is for the user to add `@Injectable` to services they want to use via DI. This is a breaking change from View Engine in the specific cases such as this one, but the impact is expected to be minimal as most services do use `@Injectable`.

alxhub reviewed (changes requested):
(see above)

pkozlowski-opensource committed:
test(ivy): @Injectable is required on type providers

pkozlowski-opensource commented:
@alxhub as discussed I've removed the traversing into providers during the JIT compilation and added `@Injectable` annotation in the test to indicate that this is required now. Please take another look and approve.

alxhub reviewed (approved):
👍
